src/transformStateWithClones.js

```diff
@@ -65,7 +65,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const transrormsHistory = [];
+  const copyState = { ...state };
+
+  for (const transform of transforms) {
+    if (transform.operation === 'addProperties') {
+      Object.assign(copyState, transform.properties);
+      transrormsHistory.push({ ...copyState });
```

Comment:
you push state 3 times in each if, but you can do it only once at the end of each iteration